src/coprocessor/dag/expr/builtin_encryption.rs

```diff
@@ -31,6 +34,37 @@ const SHA224: i64 = 224;
 const SHA256: i64 = 256;
 const SHA384: i64 = 384;
 const SHA512: i64 = 512;
+//const AES_BLOCK_SIZE: i64 = 16;
+
+fn derive_key_mysql(input: &[u8], block_size: usize) -> Vec<u8> {
+    let mut output: Vec<u8> = vec![0; block_size];
+    let mut idx = 0;
+    for k in input {
+        if idx == block_size {
+            idx = 0;
+        }
+        output[idx] ^= k;
+        idx += 1;
+    }
+    output
+}
+
+fn get_key_size(key_size: usize) -> aes::KeySize {
```

Comment:
How about implement it as a TryFrom?

Reply:
Thank you for reviewing.  Because aes::KeySize is the structure defined in crypto, if I use TryFrom I need to encapsulate aes::KeySize. Such as "struct KeySize(aes::KeySize)" . I  think it is not necessary.

Reply:
Good point. Let's keep current style.